.github/workflows/build.yml

```diff
@@ -61,132 +61,117 @@ jobs:
+          git_tag: ${{ needs.smoke-tests.outputs.GIT_TAG }}
+          event_name: ${{ github.event_name }}
+
+  build-gha-riscv64:
+    runs-on: ubuntu-latest
+    needs: smoke-tests
+    env:
+      CI_ARCH: linux/riscv64
+      GIT_BRANCH: ${{ needs.smoke-tests.outputs.GIT_BRANCH }}
+      GIT_TAG: ${{ needs.smoke-tests.outputs.GIT_TAG }}
+    steps:
       -
-        name: "Generate checksum file"
-        run: |
-          mv pihole-FTL "${{ matrix.bin_name }}"
-          sha1sum pihole-FTL-* > ${{ matrix.bin_name }}.sha1
+        name: Checkout code
+        uses: actions/checkout@v4.1.1
       -
-        name: Store binary artifacts for later deployoment
-        if: github.event_name != 'pull_request'
-        uses: actions/upload-artifact@v4.3.1
+        name: Build and test FTL
+        uses: ./.github/actions/build-and-test
         with:
-          name: ${{ matrix.bin_name }}-binary
-          path: '${{ matrix.bin_name }}*'
+          platform: linux/riscv64
+          bin_name: pihole-FTL-riscv64
+          artifact_name: riscv64-pihole-FTL-riscv64-binary
+          git_branch: ${{ needs.smoke-tests.outputs.GIT_BRANCH }}
+          git_tag: ${{ needs.smoke-tests.outputs.GIT_TAG }}
+          event_name: ${{ github.event_name }}
+
+  build-self-hosted:
+    runs-on: self-hosted
+    needs: smoke-tests
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - platform: linux/arm/v6
+            bin_name: pihole-FTL-armv6
+          - platform: linux/arm/v7
+            bin_name: pihole-FTL-armv7
+          - platform: linux/arm64/v8
+            bin_name: pihole-FTL-arm64
+    env:
+      CI_ARCH: ${{ matrix.platform }}
+      GIT_BRANCH: ${{ needs.smoke-tests.outputs.GIT_BRANCH }}
+      GIT_TAG: ${{ needs.smoke-tests.outputs.GIT_TAG }}
+    steps:
       -
-        name: Extract documentation files from container
-        if: github.event_name != 'pull_request' && matrix.platform == 'linux/amd64'
-        run: |
-          tar -xf build.tar api-docs.tar.gz
+        name: Checkout code
+        uses: actions/checkout@v4.1.1
       -
-        name: Upload documentation artifacts for deployoment
-        if: github.event_name != 'pull_request' && matrix.platform == 'linux/amd64'
-        uses: actions/upload-artifact@v4.3.1
+        name: Build and test FTL
+        uses: ./.github/actions/build-and-test
         with:
-          name: pihole-api-docs
-          path: 'api-docs.tar.gz'
+          platform: ${{ matrix.platform }}
+          bin_name: ${{ matrix.bin_name }}
+          artifact_name: ${{ matrix.bin_name }}-binary
+          git_branch: ${{ needs.smoke-tests.outputs.GIT_BRANCH }}
+          git_tag: ${{ needs.smoke-tests.outputs.GIT_TAG }}
+          event_name: ${{ github.event_name }}
 
   deploy:
     if: github.event_name != 'pull_request'
-    needs: [smoke-tests, build]
+    needs: [smoke-tests, build-gha-x86, build-self-hosted]
```

Comment:
Just reading over this file again. I'm wondering if `smoke-tests` is `need`ed at the `deploy` stage?

if `build-gha-x86` or `build-self-hosted` are true, then it stands to reason that `smoke-tests` must also be true (else they would never start)

I think I've logicked that correctly? Why isn't logicked a real word?